Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementButtonAndInformationCell.swift

```diff
@@ -0,0 +1,123 @@
+    private let labelToButtonSpacing: CGFloat = .mediumSpacing
+    private let labelLeftInset: CGFloat = .mediumLargeSpacing
+    private let labelWidthProportion: CGFloat = 0.67
+
+    private lazy var informationLabel: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.font = .title5
+        label.textColor = .licorice
+        label.textAlignment = .left
+        return label
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton(frame: .zero)
+        button.titleLabel?.font = .title5
+        button.titleLabel?.textColor = .milk
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.backgroundColor = .primaryBlue
+        button.layer.cornerRadius = 8
+        button.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+        button.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        return button
+    }()
+
+    // MARK: - init
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        selectionStyle = .none
+        button.addTarget(self, action: #selector(handleButtonTouchUpInside(_:)), for: .touchUpInside)
+        button.addTarget(self, action: #selector(handleButtonTouchDown(_:)), for: .touchDown)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - ButtonActions
+
+    @objc private func handleButtonTouchUpInside(_ sender: UIButton) {
+        delegate?.buttonAndInformationCellButtonWasTapped(self)
+
+        UIView.animate(withDuration: 0.1, delay: 0, options: .beginFromCurrentState, animations: {
+            sender.backgroundColor = .primaryBlue
+        }, completion: nil)
+    }
+
+    @objc private func handleButtonTouchDown(_ sender: UIButton) {
+        sender.backgroundColor = .callToActionButtonHighlightedBodyColor
+    }
+
+    // MARK: - Constraint updates
+
+    private func updateButtonConstraints() {
+        guard let buttonText = buttonText else { return }
+
+        let buttonWidth = 20 + buttonText.width(withConstrainedHeight: buttonHeight, font: .title5)
+        if let widthConstraint = (button.constraints.filter { $0.firstAttribute == .width }).first {
+            widthConstraint.constant = buttonWidth
+        }
+    }
+
+    private func updateInformationLabelConstraints() {
+        guard let informationText = informationText else { return }
+
+        let labelWidth = bounds.size.width*labelWidthProportion-labelToButtonSpacing-labelLeftInset
+        let labelHeight = informationText.height(withConstrainedWidth: labelWidth, font: .title5)
+        if let heightConstraint = (informationLabel.constraints.filter { $0.firstAttribute == .height }).first {
```

Comment:
Same here. Have a property and than just modify the constant instead?

Reply:
Yup, much better, agreed.